tools/wagmi-example/src/hooks/use-deploy.ts

```diff
@@ -0,0 +1,66 @@
+
+export function useDeploy() {
+  const {data: deployTxHash, deployContract, status: deployStatus} = useDeployContract()
+  const {data: deployReceipt, status: receiptStatus} = useTransactionReceipt({
+    hash: deployTxHash,
+    query: {
+      enabled: Boolean(deployTxHash),
+    }
+  })
+
+  return {
+    deployContract: () => {
+      deployContract({
+        bytecode: `0x${bin}`,
+        abi,
+        args: ["Hello from react"],
+      })
+    },
+    deployTxHash,
+    deployReceipt,
+    contractAddress: deployReceipt?.contractAddress ?? undefined,
+    isDeployed: deployStatus === "success" && receiptStatus === "success",
+  }
+}
+
+type ContractAddress = `0x${string}` | undefined;
+
+export function useWriteGreet(address: ContractAddress, refetch: () => void) {
+  const {writeContractAsync, status, data} = useWriteContract()
+
+  return {
+    setGreeting: async (greet: string) => {
+      if (!address) {
+        return
+      }
+      await writeContractAsync({
+        abi,
+        address,
+        functionName: "setGreeting",
+        args: [greet],
+      })
+      refetch()
+    },
+    status,
+    data
+  }
+}
+
+export function useGetGreet(address: ContractAddress) {
```

Comment:
Just a small suggestion: since we've wrapped use**Write**Contract into use**Write**{ContractName}  and use**Read**Contract into use**Get**{ContractName}, maybe we should stick to a single naming convention. We could either go with:

use**Get**{ContractName} and use**Set**{ContractName}

or

use**Read**{ContractName} and use**Write**{ContractName}

?